src/Language/PureScript/Make.hs

```diff
@@ -384,3 +389,34 @@ buildMakeActions outputDir filePathMap foreigns usePrefix =
 
   progress :: ProgressMessage -> Make ()
   progress = liftIO . putStrLn . renderProgressMessage
+
+-- | For a given CoreFn module and corresponding FFI module file path,
+-- check that there is exactly one FFI declaration for each
+checkForeignDecls :: CF.Module CF.Ann -> FilePath -> SupplyT Make ()
+checkForeignDecls m path = do
```

Comment:
I know this is suuuper gross with `internalError` all over the place. I just wanted to see if this approach made sense first, before worrying about the errors.